ops/log.py

```diff
@@ -87,5 +99,220 @@ def except_hook(etype: type[BaseException], value: BaseException, tb: types.Trac
         if exc_stderr:
             print(f'Uncaught {etype.__name__} in charm code: {value}', file=sys.stderr)
             print('Use `juju debug-log` to see the full traceback.', file=sys.stderr)
+        _log_security_event(
+            'WARN',
+            _SecurityEventSystem.SYS_CRASH,
+            etype.__name__,
+            description=f'Uncaught exception in charm code: {value!r}.',
+        )
 
     sys.excepthook = except_hook
+
+
+class _SecurityEventAuthN(enum.Enum):
```

Comment:
If we keep the event logging functionality as internal use only, then we definitely want to strip out all of these that we don't use (most of them).

If we do expose `log_security_event` publicly for charms to use, then I think it is helpful to have these. I don't think we want to have them in the top-level `ops` namespace - maybe they could be used from `ops.log`? We could also cut down the list to the ones that the security spec requires, rather than the full OWASP set.